src/integrationTest/java/uk/gov/hmcts/reform/sendletter/controllers/SendLetterTest.java

```diff
@@ -56,12 +59,20 @@ public void should_return_200_when_single_letter_is_sent() throws Throwable {
             .andReturn();
 
         assertThat(result.getResponse().getContentAsString()).isNotNull();
-        verify(insights).trackDependency(any(ProceedingJoinPoint.class), dependencyCaptor.capture());
-
-        Dependency dependency = dependencyCaptor.getValue();
-
-        assertThat(dependency.value()).isEqualTo(AppDependency.AUTH_SERVICE);
-        assertThat(dependency.command()).isEqualTo(AppDependencyCommand.AUTH_SERVICE_HEADER);
+        verify(insights, times(2)).trackDependency(any(ProceedingJoinPoint.class), dependencyCaptor.capture());
+
+        List<Dependency> dependencies = dependencyCaptor.getAllValues();
+        List<String> dependencyNames = dependencies.stream().map(Dependency::value).collect(toList());
+        List<String> dependencyCommands = dependencies.stream().map(Dependency::command).collect(toList());
+
+        assertThat(dependencyNames).containsExactly(
+            AppDependency.AUTH_SERVICE,
+            AppDependency.PDF_CLIENT
+        );
+        assertThat(dependencyCommands).containsExactly(
+            AppDependencyCommand.AUTH_SERVICE_HEADER,
+            AppDependencyCommand.PDF_CREATE
+        );
```

Comment:
can you just check that the `dependencies` list contains expected objects instead of extracting its' fields into separate lists?

Reply:
I am not aware of annotation interface instantiation. If exists must be non trivial

Reply:
you don't need to.

http://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#extracted-properties-assertion

```
assertThat(fellowshipOfTheRing).extracting("name", "age", "race.name")
                               .contains(
                                         tuple("Boromir", 37, "Man"),
                                         tuple("Sam", 38, "Hobbit"),
                                         tuple("Legolas", 1000, "Elf")
                               );
```

Reply:
it's a different object used in the example. annotation interfaces are different concepts. here's error output for your example ^:

```bash
Can't find any field or property with name 'value'.
Error when introspecting properties was :
- No getter for property 'value' in com.sun.proxy.$Proxy181 
```

Reply:
It crashed because there is no `getValue()` method in your class.
Try `.extractingResultOf(...)` instead (example in same link, scroll down).